test-apps/imodel-transformer/src/test/Transformer.test.ts

```diff
@@ -84,4 +85,52 @@ describe("imodel-transformer", () => {
     assert.isAtLeast(count(sourceDb, PhysicalPartition.classFullName), 2);
     targetDb.close();
   });
+
+  it("should exclude categories", async () => {
+    const targetDbFileName = initOutputFile("CompatibilityTestSeed-CategoryExcluded.bim");
+    const targetDb = SnapshotDb.createEmpty(targetDbFileName, {
+      rootSubject: { name: `${sourceDb.rootSubject.name}-CategoryExcluded` },
+      ecefLocation: sourceDb.ecefLocation,
+    });
+
+    const testSpatialCategory = {
+      name: "TestSpatialCategory",
+      id: "0x14",
+      code: new Code({
+        scope: "0x10",
+        spec: "0x16",
+        value: "TestSpatialCategory",
+      }),
+    };
+
+    await Transformer.transformAll(new BackendRequestContext(), sourceDb, targetDb, { excludeCategories: [testSpatialCategory.name] });
+
+    async function getElementsInTestCategory(db: IModelDb) {
+      const result: GeometricElementProps[] = [];
+      // do two queries because querying abstract GeometricElement won't contain the category
+      for (const type of [GeometricElement2d.classFullName, GeometricElement3d.classFullName]) {
+        for await (const elem of db.query(
+          `SELECT * FROM ${type} e JOIN bis.Category c ON e.category.id=c.ECInstanceId WHERE c.CodeValue=:category`,
+          { category: testSpatialCategory.code.value },
+        )) {
+          result.push(elem);
+        }
+      }
+      return result;
+    }
+
+    const categoryInSrc = sourceDb.elements.tryGetElement(testSpatialCategory.code);
+    assert.isDefined(categoryInSrc);
+
+    const elemsInCategoryInSrc = await getElementsInTestCategory(sourceDb);
+    assert.isAtLeast(elemsInCategoryInSrc.length, 6);
+
+    const categoryInTarget = targetDb.elements.tryGetElement(testSpatialCategory.code);
+    assert.isDefined(categoryInTarget);

```

Comment:
This is making the assumption that Code.spec and Code.scope Ids do not change between source and target. Would prefer not to make that assumption.

Reply:
I will get by code value I suppose then. I haven't fully looked into what makes two elements in different iModels considered equivalent.

Reply:
IModelTransformer uses ExternalSourceAspects to record provenance. It also maintains an IModelCloneContext that has the source element to target element mappings.